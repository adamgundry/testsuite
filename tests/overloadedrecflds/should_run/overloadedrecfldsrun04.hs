{-# LANGUAGE OverloadedRecordFields, TemplateHaskell #-}

import GHC.Records
import Language.Haskell.TH
import Language.Haskell.TH.Syntax

-- Splice in a datatype with field...
$(return [DataD [] (mkName "R") [] [RecC (mkName "MkR") [(mkName "foo", NotStrict, ConT ''Int)]] []])

-- ... and check that we can inspect it
main = do  putStrLn $(do { info <- reify ''R
                         ; lift (pprint info) })
           putStrLn $(do { insts <- reifyInstances ''Has [ConT ''R, LitT (StrTyLit "foo"), ConT ''Int]
                         ; lift (pprint insts) })
           print (foo (MkR { foo = 42 }))
