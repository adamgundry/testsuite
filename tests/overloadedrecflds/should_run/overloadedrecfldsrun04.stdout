data Main.R = Main.MkR {Main.$sel:foo:R :: GHC.Types.Int}

42
